From 13a50d064ba9f75d6088c3fcf2c673da19da4b9c Mon Sep 17 00:00:00 2001
From: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Date: Wed, 9 Jul 2014 13:18:09 +0100
Subject: [PATCH 254/351] ASoC: arizona: Move chip type check into
 arizona_wm5102_set_dac_comp

In preparation for more refactoring move the check for the chip type
into arizona_wm5102_set_dac_comp.

Change-Id: I76d448adde2ac66abbc5b7e2f681e3d77567489e
Signed-off-by: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
---
 sound/soc/codecs/arizona.c |   16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/sound/soc/codecs/arizona.c b/sound/soc/codecs/arizona.c
index f8666bf..e28dc24 100644
--- a/sound/soc/codecs/arizona.c
+++ b/sound/soc/codecs/arizona.c
@@ -1628,6 +1628,13 @@ static void arizona_wm5102_set_dac_comp(struct snd_soc_codec *codec,
 	struct arizona_priv *priv = snd_soc_codec_get_drvdata(codec);
 	struct arizona *arizona = priv->arizona;
 
+	switch (arizona->type) {
+	case WM5102:
+		break;
+	default:
+		return;
+	}
+
 	mutex_lock(&arizona->reg_setting_lock);
 	snd_soc_write(codec, 0x80, 0x3);
 	if (rate >= 176400) {
@@ -1670,14 +1677,7 @@ static int arizona_hw_params_rate(struct snd_pcm_substream *substream,
 
 	switch (dai_priv->clk) {
 	case ARIZONA_CLK_SYSCLK:
-		switch (priv->arizona->type) {
-		case WM5102:
-			arizona_wm5102_set_dac_comp(codec,
-						    params_rate(params));
-			break;
-		default:
-			break;
-		}
+		arizona_wm5102_set_dac_comp(codec, params_rate(params));
 
 		snd_soc_update_bits(codec, ARIZONA_SAMPLE_RATE_1,
 				    ARIZONA_SAMPLE_RATE_1_MASK, sr_val);
-- 
1.7.9.5

